File: runtime/flutter/cupertino/cupertinoPageScaffold.ts
import { RuntimeBaseClass } from "../../runtimeBaseClass";
import { Type } from "../../dart/core/type";
import { Color } from "../../dart/ui/color";

import { StatelessWidget } from "./../widgets/statelessWidget";
import { Key } from "../foundation/key";
import { Widget } from "../widget";
import { CupertinoNavigationBar } from "./cupertinoNavigationBar";

export interface CupertinoPageScaffoldProps {
    key?: Key | undefined;
    navigationBar?: CupertinoNavigationBar | undefined;
    backgroundColor?: Color | undefined;
    resizeToAvoidBottomInset?: boolean | undefined;
    child: Widget;
}

declare const flutter: {
    cupertino: {
        cupertinoPageScaffold: (this: void, props: CupertinoPageScaffoldProps) => CupertinoPageScaffold;
    }
}

export class CupertinoPageScaffold extends StatelessWidget implements RuntimeBaseClass {
    public readonly internalRuntimeType = new Type(CupertinoPageScaffold);
    public props: CupertinoPageScaffoldProps;
    public constructor(props: CupertinoPageScaffoldProps) {
        super();
        this.props = props;

        if (this.props.resizeToAvoidBottomInset === undefined) {
            this.props.resizeToAvoidBottomInset = true;
        }
    }

    public build() {
        return flutter.cupertino.cupertinoPageScaffold(this.props);
    }
}

File: lib/cfr/builtins/libs/flutter/cupertino/cupertinoPageScaffold.dart
import 'package:flutter/cupertino.dart';
import 'package:hydro_sdk/cfr/builtins/boxing/unboxers.dart';
import 'package:hydro_sdk/cfr/vm/closure.dart';
import 'package:hydro_sdk/cfr/vm/context.dart';
import 'package:hydro_sdk/cfr/vm/table.dart';
import 'package:hydro_sdk/hydroState.dart';

void loadCupertinoPageScaffold(
    {@required HydroState luaState, @required HydroTable table}) {
  table["cupertinoPageScaffold"] = makeLuaDartFunc(func: (List<dynamic> args) {
    return [
      CupertinoPageScaffold(
        key: maybeUnBoxAndBuildArgument<Key>(args[0]["key"],
            parentState: luaState),
        navigationBar:
            maybeUnBoxAndBuildArgument<ObstructingPreferredSizeWidget>(
                args[0]["navigationBar"],
                parentState: luaState),
        backgroundColor: maybeUnBoxAndBuildArgument<Color>(
            args[0]["backgroundColor"],
            parentState: luaState),
        resizeToAvoidBottomInset: args[0]["resizeToAvoidBottomInset"],
        child: maybeUnBoxAndBuildArgument<Widget>(args[0]["child"],
            parentState: luaState),
      )
    ];
  });
}
